src/database/UIDatabase.js

```diff
@@ -35,36 +33,53 @@ export class UIDatabase {
     this.database = database;
   }
 
+  ERRORS = {
+    ERROR_IN_TRANSACTION: { code: 'ERROR_IN_TRANSACTION', message: 'Database is in a transaction' },
+    ERROR_NO_PERMISSION: { code: 'ERROR_NO_PERMISSION', message: 'Storage permission not granted' },
+    ERROR_UNKNOWN: { code: 'ERROR_UNKOWN', message: 'ERROR_UNKNOWN' },
```

Comment:
Nit: `ERROR_UNKNOWN` typo. Shouldn't the error message be descriptive, e.g. "Unknown error occurred"?

Reply:
2666aae4249731967e787db0861901fce19e276d